scRNA-seq-advanced/setup/SCPCP000005-samples/Snakefile

```diff
@@ -0,0 +1,30 @@
+# Workflow for preprocessing SCPCP000005 data used for integration instruction
+# Raw files are expected to be present in the `config[raw_dir]` directory
+
+configfile: "config.yaml"
+outdir = "processed"
+
+rule target:
+  input:
+    expand(os.path.join(outdir, "{library}.rds"),
+           library = config["libraries"])
+
+
+
+rule prepare_library:
+  input: 
+    filtered_rds = "raw/{library}_filtered.rds",
+    celltype_tsv = config["celltypes_file"]
+  output:
+    os.path.join(outdir, "{library}.rds")
```

Comment:
Now that the input is a full path (or may be depending on config), we want to add a wildcard to be sure that snakemake knows where to look for the input files and where to put the outputs.

One thing to note here is that we don't need to include the `outdir` explicitly. It figures out where to look for the files based on the `target` now.
```suggestion
    filtered_rds = "{base_dir}/raw/{library}_filtered.rds",
    celltype_tsv = config["celltypes_file"]
  output:
    "{base_dir}/processed/{library}.rds"
```

Reply:
I accepted this change, but I'm back to getting this error (https://github.com/AlexsLemonade/training-modules/pull/580#issuecomment-1339901533) now - 
```
SyntaxError:
Not all output, log and benchmark files of rule prepare_library contain the same wildcards. This is crucial though, in order to avoid that two or more jobs write to the same file.
```

This is why I added `os.path.join()` in [`ad88d0`](https://github.com/AlexsLemonade/training-modules/pull/580/commits/ad88d07ba38cd5d22963ff7da4da8e2c208e76c3).. Any suggestions here?

Reply:
Oh, right. I ran into this before too and promptly forgot about it. The log files need to have the wildcard in them, which means they would end up in the base directory, which isn't awful, but is annoying. You will want to go back to os.path.join, but keep the `raw` and `processed` subdirectories.